docs/apworld_dev_faq.md

```diff
@@ -43,3 +43,26 @@ A faster alternative to the `for` loop would be to use a [list comprehension](ht
 ```py
 item_pool += [self.create_filler() for _ in range(total_locations - len(item_pool))]
 ```
+
+---
+
+### I learned about indirect conditions in the world API document, but I want to know more. What are they and why are they necessary?
+
+The world API document mentions indirect conditions and **when** you should use them, but not *how* they work and *why* they are necessary. This is because the explanation is quite complicated.
+
+Region sweep (the algorithm that determines which regions are reachable) is a Breadth-First Search of the region graph from the origin region, checking entrances one by one and adding newly reached nodes (regions) and their entrances to the queue until there is nothing more to check.
+
+For performance reasons, AP only checks every entrance once. However, if entrance access conditions depend on regions, then it is possible for this to happen:
```

Comment:
rewriting the suggestion because it technically had a conflict lol
```suggestion
For performance reasons, AP only checks every entrance once. However, if an entrance's access condition depends on regions, then it is possible for this to happen:
```
```suggestion
For performance reasons, AP only checks every entrance once. However, if any entrance access condition depends on regions, then it is possible for this to happen:
```
```suggestion
For performance reasons, AP only checks every entrance once. However, if entrance Access Conditions depend on regions, then it is possible for this to happen:
```
```suggestion
For performance reasons, AP only checks every entrance once. However, if entrance access_conditions depend on regions, then it is possible for this to happen:
```
giving a couple more options too, because I don't think there's one way to fix this, but as-is it reads strange without the assumption that "access condition" is one idea and not multiple words you have to parse their role in the sentence
easiest is probably to rewrite the whole sentence, but i think any of these help 😅